tests/Models/Http/HeadersTest.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Infrastructure\Tests\Models\Http;
+
+use Infrastructure\Models\Http\Headers;
+use Infrastructure\Models\Http\IllegalHeaderValueException;
+use Infrastructure\Tests\ReflectionHelperTrait;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class HeadersTest
+ * @see Headers
+ */
+class HeadersTest extends TestCase
+{
+    use ReflectionHelperTrait;
+
+    /**
+     * @see Headers::hasHeader()
+     *
+     * @throws \ReflectionException
+     */
+    public function testHasHeader()
+    {
+        $data = [
+            'X-TEST-1' => ['val_1', 'val_2'],
+            'X-TEST-2' => ['val_3', 'val_4'],
+            'X-TEST-3' => ['val_5'],
+        ];
+        /** @var Headers $headers */
+        $headers = $this->createInstance(Headers::class);
```

Comment:
What is the reason create to  object so weird ? 

What is main purpose of this test? 

What is diff if you create object in natural way? 

If __construct doesn't work your test will fail which is correct. 

This code is redundant and hiding normal behavior.